src/custom/components/swap/EthWethWrap/index.tsx

```diff
@@ -81,33 +83,55 @@ const BalanceLabel = styled.p<{ background?: string }>`
   }
 `
 
+const ErrorWrapper = styled(BalanceLabel)`
+  font-size: x-small;
+  background-color: #ff000040;
+  color: ${({ theme }) => theme.red1};
+`
+
+const ErrorMessage = ({ error }: { error: Error }) => (
+  <ErrorWrapper>
+    <i>
+      <strong>{error.message}</strong>
+    </i>
+  </ErrorWrapper>
+)
+
 export interface Props {
   account?: string
   native: Currency
   wrapped: Token
-  wrapCallback: () => Promise<void>
+  wrapCallback: () => Promise<TransactionResponse>
 }
 
 export default function EthWethWrap({ account, native, wrapped, wrapCallback }: Props) {
   const [loading, setLoading] = useState(false)
   const [error, setError] = useState<Error | null>(null)
+  const [pendingHash, setPendingHash] = useState<string | undefined>()
+
+  const isWrapPending = useIsTransactionPending(pendingHash)
```

Comment:
Mm, i see how you wanted to implement this, but the problem of this is that `pendingHash` is in the local state of the component. 

One simple solution is that we create a trivial hook: `useIsWrapping(): boolean`

This hook delegates on `useAllTransactions()` and returns true if ANY tx is of wrapping. 

This has 2 advantages:
- If the user is wrapping because he chosed ETH/WETH and wrap, and the tx is mined, we will see it as loading. So he will know is still unmined
- If the user change the selection of the token (for example clicking the big orange button), and then select ETH again before the tx is mined, he will be able to do a second WRAP although the first one is still unmined

Reply:
@anxolin I see the benefit of the external hook as you mention but I don't understand the `advantages` part:
1. why would the user see anything loading if a wrap tx is mined? if it's mined it has returned us a receipt and the pending tx is no longer pending
2. with the new hook, the point is that he/she WON'T be able to wrap again because the new hook will return the wrapping tx as pending if it hasn't been mined - or are you saying that's the current behavour as it resets local state?

Reply:
yep, sorry I meant the points as disadvantages of the current approach. I'll edit my comment

> why would the user see anything loading if a wrap tx is mined? if it's mined it has returned us a receipt and the pending tx is no longer pending

I'll edit my comment. I meant `unmined` in the example

>  or are you saying that's the current behavour as it resets local state?

yep, this is what i tried to say

Reply:
done in #473